### PR TITLE
Adding quickstart page and updating README with the same content

### PR DESCRIPTION
**Pull request by e-staats**
Hey Josh, I love the app and thought it could use a Quickstart guide for those who want to get an idea of how to use Spiel. Feel free to edit as you see fit!

Tasks
-----

- [x] Updated changelog (didn't change any code, just updated documentation)
- [x] Updated documentation.

### CODE REVIEW

**Comment by JoshKarpel** [comment]
Nice, thanks for doing this! I'll try to find some time to review soon.

File: README.md
Comment (JoshKarpel) [inline README.md:41]:
```suggestion
    return "Your content here!"
```
let's make this as simple as humanly possible

File: README.md
Comment (JoshKarpel) [inline README.md:47]: Let's explicitly note how to do this, e.g. something like "put the above code in `deck.py` then run `python deck.py`"

File: README.md
Comment (JoshKarpel) [inline README.md:48]: I was playing with these over the last few days - I should really write a contribution guide to describe how to do this...

There's a script at https://github.com/JoshKarpel/spiel/blob/30e676bee4bed388eaaa8901cc41f734b1ecc2d6/docs/generate_screenshots.py#L62-L64 that is hooked up to `pre-commit` (incidentally, make sure to install `pre-commit` and then run `pre-commit install`). The idea is to make sure all the screenshots are always up-to-date and to identify any changes elsewhere that cause those screenshots to change. So, I'd prefer if you could refactor that script so that it can work with a deck that isn't the demo deck, and to generate these screenshots via that script.

Also note how I did the URLs in https://github.com/JoshKarpel/spiel/pull/174/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R24-R25 - that should make them work on PyPI next time I release 🤞🏻 (this `README.md` becomes the PyPI project description).

File: README.md
Comment (JoshKarpel) [inline README.md:50]:
```suggestion
To recap, you first create a `Deck` that has the name of your presentation.
Then you create slide functions with the `@deck.slide()` decorator.
The slide function should return anything that [Rich can render](https://rich.readthedocs.io/en/stable/console.html#printing) - that return value will be displayed as the slide in the presentation.
Finally, you call `present()` to run the presentation.
```

Some content changes, plus I like one slide/clause per line to make diffs easier to read.

File: README.md
Comment (JoshKarpel) [inline README.md:52]: Can probably drop this with the change I suggested above since I mention Rich earlier, though I do like this line, maybe work it back in somehow?

File: docs/quickstart.md
Comment (JoshKarpel) [inline docs/quickstart.md:72]: One that that I'd love to make it into this is the `spiel demo copy/source/present` commands https://github.com/JoshKarpel/spiel/blob/f503926c467d586a21a16b1c931759374ae7988b/spiel/cli.py#L131-L192

File: docs/quickstart.md
Comment (JoshKarpel) [inline docs/quickstart.md:66]: Executable code examples was actually pretty tricky to implement for https://github.com/JoshKarpel/pytest-its-whats-for-testing and I would consider it (currently) a non-standard feature 😅 . Though it's such an obvious direction to go with the tool that I should probably figure out a way to roll it in first-class...

Point is - maybe some other example here? I've always thought that this https://github.com/JoshKarpel/spiel/blob/6321c01418a0d0b262042ef9a555e652803c4772/spiel/demo/demo.py#L122-L156 demo slide was a good simple example, showing off both fancy Rich rendering with layouts and also the introspection capabilities of running inside Python.

File: docs/quickstart.md
Comment (JoshKarpel) [inline docs/quickstart.md:1]:
```suggestion
# Quick Start
```

File: docs/quickstart.md
Comment (JoshKarpel) [inline docs/quickstart.md:3]: I *might* prefer if we just linked here from the `README.md` to de-duplicate this chunk. Maybe the `README.md` should be very code-example-y? Just a chunk of sample code and the resulting output with a line or two of text, then link here?

**Review by JoshKarpel — CHANGES_REQUESTED** [review]

File: docs/quickstart.md
Comment (e-staats) [reply docs/quickstart.md:3]: I personally tend to like it when a README gives me an overview of what to do without having to click off into another page. It is a little clumsy to have the same content copy/pasted both places, so I'll let you make the final call on what the README should have vs. link out to!

File: README.md
Comment (e-staats) [reply README.md:48]: I made some updates in my branch, but I'm failing the pre-commit because "TypeError: SpielApp.__init__() got an unexpected keyword argument 'show_messages'"

Everything looks fine, but it's failing for me in the main pre-commit too, so I don't think it was introduced by my changes. Any ideas what might be going wrong there?

File: docs/quickstart.md
Comment (JoshKarpel) [reply docs/quickstart.md:3]: Fair enough! I'll think about this some more and make the final edits on this point before merging.

File: README.md
Comment (JoshKarpel) [reply README.md:48]: Interesting, I can't reproduce, but presumably it's failing inside `generate_screenshots.py`? I suspect the problem is that you might have `spiel` installed via `pip` and it's overriding the local install. I recommend redoing your `poetry install` and making sure you're in `poetry shell`. I should really write some contributing docs...

Comment (e-staats) [reply README.md:48]: Hmm, yeah it's failing in `generate_screenshots.py`. I wasn't using poetry, just pip install and a virtual environment. I installed poetry, but I'm not super familiar with it or what I should be doing differently. Is it as simple as `poetry shell` and then `git commit`? Or are there other things I should be doing in there?

Comment (JoshKarpel) [reply README.md:48]: Pretty much! I threw together a quick guide here https://www.spiel.how/contributing/#initial-setup ; feel free to edit that too if it doesn't work as expected.

Comment (e-staats) [reply README.md]: Thanks, that worked!

**Comment by e-staats** [comment]
I addressed all the outstanding comments!

**Comment by JoshKarpel** [comment]
> I addressed all the outstanding comments!

Awesome, thank you so much! I did a pass myself, will merge soon.

**Review by JoshKarpel — APPROVED** [review]

**Comment by JoshKarpel** [comment]
Hrumph, something is up with the code slide rendering in CI. Will hack at it when I'm more awake :)

**Pull request merged**
